hq/app/logic/IamRemediation.scala

```diff
@@ -106,7 +106,24 @@ object IamRemediation extends Logging {
     * This might fail, because it may be that no matching key exists.
     */
   def lookupCredentialId(badKeyCreationDate: DateTime, userCredentials: List[CredentialMetadata]): Attempt[CredentialMetadata] = {
-    ???
+    userCredentials.filter { credentialMetadata =>
+      credentialMetadata.creationDate.withTimeAtStartOfDay() == badKeyCreationDate.withTimeAtStartOfDay()
```

Comment:
If I'm reading this correctly then the edge case is a little broader because the date only has to  be on the same _day_ rather than the same _second_.

I would say though that (coincidentally) if 2 keys have the same creation date, and one of them is vulnerable, then the other might be as well. I haven't fully considered whether this is always true - because we probably can't rely on it here anyway, but thought I'd mention

Reply:
You make a good point @jfsoul. I've removed the `.withTimeAtStartOfDay` and added a test to check that if the dates do not match to the second, a failure is returned.